pontoon/sync/management/commands/import_pootle.py

```diff
@@ -0,0 +1,153 @@
+            'marketplace-marketplace': 'marketplace',
+            'marketplace-commbadge': 'commbadge',
+            'marketplace-fireplace': 'fireplace',
+            'marketplace-spartacus': 'spartacus',
+            'marketplace-stats': 'marketplace_stats',
+            'marketplace-zippy': 'zippy',
+            'master-firefox-os': 'masterfirefoxos',
+            'mdn': 'mdn',
+            'mozillians': 'mozillians',
+            'social-api-directory': 'socialapi-directory',
+            'sumo': 'sumo',
+        }.get(project)
+
+        if not pootle_project:
+            return
+
+        # Get Pootle data
+        filename = pootle_locale + '_' + pootle_project + '.json'
+        url = 'http://svn.mozilla.org/projects/l10n-misc/trunk/pontoon/verbatim/' + filename
+
+        try:
+            pootle = requests.get(url).json()
+        except ValueError:
+            return self.stdout.write("--- Skipping missing file {}.".format(filename))
+
+        # Get all Pontoon translations matching Pootle data
+        paths = pootle.keys()
+        entities = []
+        strings = []
+        emails = []
+
+        for path in paths:
+            for entity in pootle[path].keys():
+                entities.append(entity)
+                strings.append(pootle[path][entity]["translation"])
+                emails.append(pootle[path][entity]["author"])
+
+        translations = Translation.objects.filter(
+            entity__resource__project__slug=project,
+            entity__resource__path__in=paths,
+            entity__string__in=entities,
+            locale__code=locale,
+            user=None,
+            string__in=strings
+        )
+
+        # Save user permissions (and evaluate users queryset)
+        users = User.objects.filter(email__in=emails)
+        missing_users_list = []
+
+        for u in users:
+            if not u.has_perm('base.can_localize'):
+                can_localize = Permission.objects.get(codename="can_localize")
+                u.user_permissions.add(can_localize)
+                self.stdout.write("Permission granted to user {}.".format(u.email))
+
+        if users:
+            bulk_update(users)
+
+        for t in translations:
+            try:
+                pootle_translation = pootle[t.entity.resource.path][t.entity.string]
+
+                # Save user
+                email = pootle_translation["author"]
+                user = users.get(email=email)
+                t.user = user
+                t.approved_user = user
+
+                # Save date
+                date = pootle_translation["date"].split("+")[0]
+                dateObj = timezone.make_aware(datetime.strptime(date, "%Y-%m-%d %H:%M:%S"))
+                t.date = dateObj
+                t.approved_date = dateObj
+
+            except User.DoesNotExist:
+                missing_users_list.append(email)
+
+            # Path - entity mismatch, skip to next translation
+            except KeyError:
```

Comment:
You'll want to move this up to only surround the `pootle_translation` line so you don't catch any unexpected errors, and also saves you some indenting. :D